File: CSharp/ProjectEuler2.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.InteropServices;
using System.Text;
using System.Threading.Tasks;

namespace TestSC
{
    class Program
    {
        public static IEnumerable<int> FibSeq()
        {
            int firstNumber = 1;
            int secondNumber = 0;
            
            yield return secondNumber;

            while (firstNumber < 40000000)
            {
                yield return firstNumber;
                int tmp = firstNumber;
                firstNumber += secondNumber;
                secondNumber = tmp;
            }
        }
        static void Main(string[] args)
        {
            var sum = FibSeq().Where(x=>x % 2 == 0).Sum(); 

            Console.WriteLine(sum);
        }
    }
}

File: kotlin/ProjectEuler2.kt

class Main {
    fun fibSeq(condition:Int):List<Int>
    {
        var firstNumber = 1;
        var secondNumber = 0;
        var res = arrayListOf(secondNumber)

        while (firstNumber < condition)
        {
            res.add(firstNumber)
            val tmp = firstNumber;
            firstNumber += secondNumber;
            secondNumber = tmp;
        }
        return res
    }
    fun main(args : Array<String>) {
        val quantity = 40000000;
        val res = fibSeq(quantity).filter { x -> x % 2 == 0 }.sum()
        println(res)

    }
}
